==3018== Memcheck, a memory error detector
==3018== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3018== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3018== Command: ./gradebook_main
==3018== Parent PID: 2908
==3018== 
==3018== Invalid read of size 1
==3018==    at 0x484FBD4: strcmp (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3018==    by 0x109469: add_score (gradebook.c:47)
==3018==    by 0x109CA2: main (gradebook_main.c:72)
==3018==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==3018== 
==3018== 
==3018== Process terminating with default action of signal 11 (SIGSEGV)
==3018==  Access not within mapped region at address 0x0
==3018==    at 0x484FBD4: strcmp (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3018==    by 0x109469: add_score (gradebook.c:47)
==3018==    by 0x109CA2: main (gradebook_main.c:72)
==3018==  If you believe this happened as a result of a stack
==3018==  overflow in your program's main thread (unlikely but
==3018==  possible), you can try to increase the size of the
==3018==  main thread stack using the --main-stacksize= flag.
==3018==  The main thread stack size used in this run was 8388608.
==3018== 
==3018== HEAP SUMMARY:
==3018==     in use at exit: 16,048 bytes in 3 blocks
==3018==   total heap usage: 3 allocs, 0 frees, 16,048 bytes allocated
==3018== 
==3018== 1,024 bytes in 1 blocks are still reachable in loss record 1 of 3
==3018==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3018==    by 0x48DEBA3: _IO_file_doallocate (filedoalloc.c:101)
==3018==    by 0x48EDCDF: _IO_doallocbuf (genops.c:347)
==3018==    by 0x48ECF5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==3018==    by 0x48EB6D4: _IO_new_file_xsputn (fileops.c:1243)
==3018==    by 0x48EB6D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==3018==    by 0x48E0F1B: puts (ioputs.c:40)
==3018==    by 0x109A17: main (gradebook_main.c:20)
==3018== 
==3018== 1,024 bytes in 1 blocks are still reachable in loss record 2 of 3
==3018==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3018==    by 0x48DEBA3: _IO_file_doallocate (filedoalloc.c:101)
==3018==    by 0x48EDCDF: _IO_doallocbuf (genops.c:347)
==3018==    by 0x48ECCDB: _IO_file_underflow@@GLIBC_2.2.5 (fileops.c:485)
==3018==    by 0x48EDD95: _IO_default_uflow (genops.c:362)
==3018==    by 0x48C30CF: __vfscanf_internal (vfscanf-internal.c:628)
==3018==    by 0x48C2141: __isoc99_scanf (isoc99_scanf.c:30)
==3018==    by 0x109ADF: main (gradebook_main.c:38)
==3018== 
==3018== 14,000 bytes in 1 blocks are still reachable in loss record 3 of 3
==3018==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3018==    by 0x10939D: create_gradebook (gradebook.c:20)
==3018==    by 0x109B8F: main (gradebook_main.c:53)
==3018== 
==3018== LEAK SUMMARY:
==3018==    definitely lost: 0 bytes in 0 blocks
==3018==    indirectly lost: 0 bytes in 0 blocks
==3018==      possibly lost: 0 bytes in 0 blocks
==3018==    still reachable: 16,048 bytes in 3 blocks
==3018==         suppressed: 0 bytes in 0 blocks
==3018== 
==3018== For lists of detected and suppressed errors, rerun with: -s
==3018== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
